homeassistant/components/vesync/light.py

```diff
@@ -175,3 +180,165 @@ def color_temp(self) -> int:
         )
         # ensure value between minimum and maximum Mireds
         return max(self.min_mireds, min(color_temp_value, self.max_mireds))
+
+
+class VeSyncMulticolorLightHA(VeSyncTunableWhiteLightHA, LightEntity):
+    """Representation of a VeSync Multicolor Light device."""
+
+    _attr_supported_color_modes = {ColorMode.HS, ColorMode.COLOR_TEMP}
+
+    @property
+    def color_mode(self) -> ColorMode | str | None:
+        """Return the current color mode."""
+        # get value from pyvesync library api,
+        color_mode_value = self.device.color_mode
+        if color_mode_value.lower() == "white":
+            return ColorMode.COLOR_TEMP
+        if color_mode_value.lower() in ["hsv", "color"]:
+            return ColorMode.HS
+        return ColorMode.ONOFF
+
+    @property
+    def hs_color(self) -> tuple[float, float] | None:
+        """Return the current color mode."""
+        # get value from pyvesync library api,
+        color_value_hsv = None
+        try:
+            # get hsv named tuple from pyvesync library (hue, sat, value)
+            color_value_hsv = self.device.color_value_hsv
+        except ValueError:
+            # deal if any unexpected/non numeric value
+            _LOGGER.debug(
+                "VeSync - received unexpected 'color_value_hsv' value "
+                "from pyvesync api: %s",
+                str(color_value_hsv),
+            )
+            return None
+        hs_color = color_value_hsv.hue, color_value_hsv.saturation
+        return hs_color
+
+    @property
+    def brightness(self) -> int | None:
+        """Get light brightness."""
+        # If Color Mode = COLOR use HSV value
+        if self.color_mode == ColorMode.HS:
+            # get value from pyvesync library api,
+            color_value_hsv = None
+            try:
+                # get hsv named tuple from pyvesync library (hue, sat, value)
+                color_value_hsv = self.device.color_value_hsv
```

Comment:
Maybe I did that because calling the library ( which in turn would call the cloud api ) could get a timeout from the cloud, and then the library would return an error log message instead of the expected number value. I don't really remember right, but I can double check. But it seems that this portion of code was done for robustness so it wouldn't break easily.
To be honest, I've never seen that debug message in my homeassistant logs. So maybe, once all the code got mature enough the issues I had while I was developing disappeared.

Reply:
OK, then add a comment explaining why this is within the try-clause